tck/features/LabelsAcceptance.feature

```diff
@@ -0,0 +1,264 @@
+      | []           |
+    And the side effects should be:
+      | +nodes | 1 |
+
+  Scenario: Creating node with two labels
+    When executing query:
+      """
+      CREATE (node:Foo:Bar {name: 'Mattias'})
+      RETURN labels(node)
+      """
+    Then the result should be:
+      | labels(node)   |
+      | ['Foo', 'Bar'] |
+    And the side effects should be:
+      | +nodes      | 1 |
+      | +labels     | 2 |
+      | +properties | 1 |
+
+  Scenario: Ignore space when creating node with labels
+    When executing query:
+      """
+      CREATE (node :Foo:Bar)
+      RETURN labels(node)
+      """
+    Then the result should be:
+      | labels(node)   |
+      | ['Foo', 'Bar'] |
+    And the side effects should be:
+      | +nodes  | 1 |
+      | +labels | 2 |
+
+  Scenario: Create node with label in pattern
+    When executing query:
+      """
+      CREATE (n:Person)-[:OWNS]->(x:Dog)
+      RETURN labels(n)
+      """
+    Then the result should be:
+      | labels(n)  |
+      | ['Person'] |
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 1 |
+      | +labels        | 2 |
+
+  Scenario: Fail when adding new label predicate on already bound node 1
+    When executing query:
+      """
+      CREATE (n: Foo)-[:T1]->(b), (n: Bar)-[:T2]->(c)
+      """
+    Then a SyntaxError should be raised at compile time: VariableAlreadyBound
+
+  Scenario: Fail when adding new label predicate on already bound node 2
+    When executing query:
+      """
+      CREATE (c)<-[:T2]-(n: Foo), (n: Bar)<-[:T1]-(b)
+      """
+    Then a SyntaxError should be raised at compile time: VariableAlreadyBound
+
+  Scenario: Fail when adding new label predicate on already bound node 3
+    When executing query:
+      """
+      CREATE (n :Foo) CREATE (n :Bar)-[:OWNS]->(x:Dog)
+      """
+    Then a SyntaxError should be raised at compile time: VariableAlreadyBound
+
+  Scenario: Fail when adding new label predicate on already bound node 4
+    When executing query:
+      """
+      CREATE (n {}) CREATE (n :Bar)-[:OWNS]->(x:Dog)
+      """
+    Then a SyntaxError should be raised at compile time: VariableAlreadyBound
+
+  Scenario: Fail when adding new label predicate on already bound node 5
+    When executing query:
+      """
+      CREATE (n :Foo) CREATE (n {})-[:OWNS]->(x:Dog)
+      """
+    Then a SyntaxError should be raised at compile time: VariableAlreadyBound
+
+  Scenario: Add labels inside `FOREACH`
+    When executing query:
+      """
+      CREATE (a), (b), (c)
+      WITH [a, b, c] AS nodes
+      FOREACH(n IN nodes |
+        SET n :Foo:Bar
+      )
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes  | 3 |
+      | +labels | 6 |
+    When executing control query:
+      """
+      MATCH (n)
+      WHERE NOT(n:Foo AND n:Bar)
+      RETURN n
+      """
+    Then the result should be:
+      | n |
+
+  Scenario: Using `labels()` in return clauses
```

Comment:
In this case they make the test name not become super weird in the jUnit runner, due to the presence of parentheses. I think we can live with them for function names like this.